client/src/components/TinymceInlineToolbar/TinymceInlineToolbar.js

```diff
@@ -0,0 +1,243 @@
+    const topOffset = 10;
+    let className = '';
+    let top = 0;
+    let left = 0;
+
+    if (spaceTop >= editorHeight || spaceBottom >= editorHeight) {
+      return this.hide();
+    }
+
+    if (this.bottom) {
+      if (spaceBottom >= spaceNeeded) {
+        className = ' mce-arrow-up';
+        top = selection.bottom + iframeRect.top + scrollY + topOffset;
+      } else if (spaceTop >= spaceNeeded) {
+        className = ' mce-arrow-down';
+        top = (selection.top + iframeRect.top + scrollY) - toolbarHeight - margin;
+      }
+    } else if (spaceTop >= spaceNeeded) {
+      className = ' mce-arrow-down';
+      top = (selection.top + iframeRect.top + scrollY) - toolbarHeight - margin;
+    } else if (
+      spaceBottom >= spaceNeeded &&
+      editorHeight / 2 > (selection.bottom + iframeRect.top) - blockedTop
+    ) {
+      className = ' mce-arrow-up';
+      top = selection.bottom + iframeRect.top + scrollY + topOffset;
+    }
+
+    if (typeof top === 'undefined') {
+      top = scrollY + blockedTop + buffer;
+    }
+
+    left = (selectionMiddle - (toolbarWidth / 2)) + iframeRect.left + scrollX;
+
+    if (selection.left < 0 || selection.right > iframeRect.width) {
+      left = iframeRect.left + scrollX + ((iframeRect.width - toolbarWidth) / 2);
+    } else if (toolbarWidth >= windowWidth) {
+      className += ' mce-arrow-full';
+      left = 0;
+    } else if (
+      (left < 0 && selection.left + toolbarWidth > windowWidth) ||
+      (left + toolbarWidth > windowWidth && selection.right - toolbarWidth < 0)
+    ) {
+      left = (windowWidth - toolbarWidth) / 2;
+    } else if (left < iframeRect.left + scrollX) {
+      className += ' mce-arrow-left';
+      left = selection.left + iframeRect.left + scrollX;
+    } else if (left + toolbarWidth > iframeRect.width + iframeRect.left + scrollX) {
+      className += ' mce-arrow-right';
+      left = (selection.right - toolbarWidth) + iframeRect.left + scrollX;
+    }
+
+    toolbarEl.className = toolbarEl.className.replace(/ ?mce-arrow-[\w]+/g, '') + className;
+    this.setStyles({ left, top });
+
+    return this;
+  }
+}
+
+/**
+ * @param  tinymce selection object
+ * @param  {Boolean}
+ * @param  {Array} array of tag names - e.g. "A"
+ */
+function shouldShowToolbar(selection, isEditorFocused, tagTypes) {
+  return selection &&
+    tagTypes.map(tag => tag.toLowerCase())
+      .includes(selection.tagName.toLowerCase()) &&
+    isEditorFocused;
+}
+
+
+/**
+ * @param  {Object} tinymce editor instance
+ * @param  {Array} array of button literal object
+ * @param  {Array} array of tag names - e.g. "A"
+ */
+function setupTinyMceInlineToolbar(editor, buttons, tagTypes = ['a']) {
+  const toolbar = new TinymceInlineToolbar(editor, buttons);
+  let currentSelection = null;
+  let scrollTimeoutId = null;
+  let focused = false;
+
+  toolbar
+    .hide()
+    .renderTo(window.document.body);
+
+  toolbar.setStyles({ position: 'absolute' });
```

Comment:
could this be part of the css file?
maybe not, it's a little bit difficult to track